src/main/kotlin/com/openlattice/assembler/processors/AddMembersToOrganizationAssemblyProcessor.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2019. OpenLattice, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@openlattice.com
+ *
+ *
+ */
+
+package com.openlattice.assembler.processors
+
+import com.hazelcast.core.Offloadable
+import com.hazelcast.spi.ExecutionService
+import com.kryptnostic.rhizome.hazelcast.processors.AbstractRhizomeEntryProcessor
+import com.openlattice.assembler.AssemblerConnectionManager
+import com.openlattice.assembler.OrganizationAssembly
+import com.openlattice.assembler.PostgresDatabases
+import com.openlattice.organizations.PrincipalSet
+import org.slf4j.LoggerFactory
+import java.lang.IllegalStateException
+import java.util.*
+
+private val logger = LoggerFactory.getLogger(AddMembersToOrganizationAssemblyProcessor::class.java)
+private const val NOT_INITIALIZED = "Assembler Connection Manager not initialized."
+
+data class AddMembersToOrganizationAssemblyProcessor(val newMembers: PrincipalSet)
+    : AbstractRhizomeEntryProcessor<UUID, OrganizationAssembly, Void?>(), Offloadable {
+
+    @Transient
+    private var acm: AssemblerConnectionManager? = null
+
+    override fun process(entry: MutableMap.MutableEntry<UUID, OrganizationAssembly?>): Void? {
+        val organizationId = entry.key
+        val assembly = entry.value
+        if (assembly == null) {
+            logger.error("Encountered null assembly while trying to add new members $newMembers  to organization " +
```

Comment:
is this expected whereas `acm == null` is not? Why does this one only log an error and the other null case throws an exception?

Reply:
In general: Acm needs to be initialized with init() func first to be able to use the processor. The assembly on the other hand can be null, if the organization(id) is not yet in the HazelcastMap.
In this case and in case of some other processors, it is maybe worth to throw an error, since we expect the organizationassembly to be already initialized, I followed the logic on the existing processors on this one.

Reply:
Assembly can be null, but this case should never happen as when organization is created its assembly is initialized. It's probably fine to throw an exception here but is not a fatal error. 

The ACM not being initialized on the other hand is a fatal error and should cause things to explode.